### PR TITLE
Fixes #418 Modify Travis to enable Codecov

### LINKED ISSUE #418: Modify Travis to enable Codecov
**I'm submitting a ...**
- [x] bug report
- [ ] feature request

**Current behavior:**

Currently, the `codecov` having problems in `.travis.yml`. 
**Expected behavior:**

```
after_success:
        - bash <(curl -s https://codecov.io/bash)

```
might solve the issue.
**Steps to reproduce:**

**Yes, I would like to work on this issue**

### PR DESCRIPTION
**Pull request by bhaveshAn**
Fixes #418 

#### Checklist

- [x] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia) and my PR follows them.
- [x] My branch is up-to-date with the Upstream `master` branch.
- [x] I have added necessary documentation (if appropriate)

#### Changes proposed in this pull request:

- Modifying `after_success` attribute of travis to enable CodeCov

@vaibhavsingh97 @AnshulMalik @mariobehling @niranjan94 Please review

### CODE REVIEW

**Comment by AnshulMalik** [comment]
Got some points
- Why do we need codecov?
- If we do need this, you should provide a sample example that we can iterate from.

**Comment by bhaveshAn** [comment]
> Why do we need codecov?

Codecov is a reporting tool that is intended to process any coverage report format into a format that is standard across Codecov. You may change the configuration of how Codecov processes reports and expresses coverage information.

The Codecov pull request comment is a highly detailed overview of a pull request. It provides details on the coverage changes in the pull request that help with speeding up code review, and ensuring that pull requests are well tested. The comment is customizable to fit your desired needs.

> If we do need this, you should provide a sample example that we can iterate from.

Checkout the PRs of https://github.com/fossasia/susper.com/pulls and https://github.com/fossasia/open-event-server/pulls for example. Here projects uses coverage changes in the pull request that help with speeding up code review,

**Comment by cclauss** [comment]
An alternative approach... https://github.com/fossasia/query-server/pull/421/files

**Comment by bhaveshAn** [comment]
Both the approaches are different 

> An alternative approach... https://github.com/fossasia/query-server/pull/421/files

It says to add the codecov reports of `tests/pytest` only.

But here, adding codecov in `after_success` helps us to get the codecov coverage of overall codebase which is changed by the contributor in the respective PR. And that is what following means: 
```
The Codecov pull request comment is a highly detailed overview of a pull request. 
It provides details on the coverage changes in the pull request that help with speeding up code review, and ensuring that pull requests are well tested. 
The comment is customizable to fit your desired needs.
```

**Comment by cclauss** [comment]
Would adding __--cov=codecov__ onto the end of __pytest --capture=sys --cov=codecov__ help or not?

**Comment by cclauss** [comment]
Should the codecov report appear in this "conversation" as it does in the example projects above?

**Comment by bhaveshAn** [comment]
Added Codecov token (`codecov.yml`) and Codecov button (which tells status) in `README.md` as well.

@mariobehling Please review

File: codecov.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+codecov:
```
Comment (vaibhavsingh97) [inline codecov.yml]: `Codecov pull request comment is a highly detailed overview of a pull request`. is current config able to do that? As I don't think so

Comment (bhaveshAn) [reply codecov.yml]: I have generated the codecov token from https://codecov.io/gh/fossasia/query-server/settings
And only a Maintainer can access that token, as he will have push rights.

Comment (bhaveshAn) [reply codecov.yml]: @vaibhavsingh97 
screenshot from 2018-01-08 18-00-22

**Comment by bhaveshAn** [comment]
@vaibhavsingh97 @mariobehling Please review

**Comment by cclauss** [comment]
What is holding up this PR?  Can we get some positive review here?

Codecov adds value to our testing ad demonstrated in other Fossasia projects.  This is unlike "Hound" which has been asleep for six months and the misconfigured Codacy that actively prevents us from adding by-the-book pytests to our codebase.

**Comment by vaibhavsingh97** [comment]
@bhaveshAn As an test can you please send dummy PR and check the functionality. Also, if its not working then please open issue and fix the same. Thanks.

**Pull request merged**